Fixes for android and llvm

jiangenj opened the pull request:
- in android file needs to be aboslute path
- add EXCLUDE_PATHS var to bypass multiple defination error
- use readelf instead of objdump to get section offset and size

ianamason commented:
What’s with the copyright lines?

jiangenj commented:
> What’s with the copyright lines?

I'm from quicinc, that's what the company asked for.
I'm fine to remove it if you insist to not have any copyright, but I think license is more important here.

ianamason commented:
I'm not keen on the copyright.
It would be nice if you could summarize the changes. 
Is that a new environment variable I see? That will need some explanation in the README to be of any use.
Do these changes effect other platforms other than android?

jiangenj commented:
> I'm not keen on the copyright. It would be nice if you could summarize the changes. Is that a new environment variable I see? That will need some explanation in the README to be of any use. Do these changes effect other platforms other than android?

- updated the readme for the var.
- updated the summary of the pr.
- and there is no side effect as readelf has consistant output with gnu or llvm ones while objdump diffs.

ianamason commented on wllvm/extraction.py:
This breaks the cross compilation feature. Read the README more carefully.

jiangenj replied:
No, it won't break it, 
```
cmd_hypcopy = $(OBJCOPY) --prefix-symbols=__kvm_nvhe_ $< $@
```
is from linux kernel, extract-bc doesn't support the --prefix-symbols, so we have to use EXCLUDE_PATHS to exclude some bc from llvm-link.

ianamason commented on wllvm/filetype.py:
Why do we want to hard code /usr/bin as the location of file? What if its in /usr/local/bin?

jiangenj replied:
use abs path first

jiangenj commented on wllvm/filetype.py:
```diff
@@ -34,7 +35,7 @@ def getFileType(cls, fileName):
         """
         retval = None
-        fileP = Popen(['file', os.path.realpath(fileName)], stdout=PIPE)
+        fileP = Popen(['/usr/bin/file', os.path.realpath(fileName)], stdout=PIPE)
```
so in Android build system, host tool can not be used as PATH var is reset. let me add a check here for /usr/bin/file or /usr/local/bin/file

ianamason replied:
So I'm not going to merge in hard coded paths like this. If you look at the code base you will notice that most tools we use (via Popen) have the feature that you can specify the full path via an environment variable.  If it was well documented I would include this approach. But not a hardcoded path hack.

jiangenj commented:
can u review again?

ianamason commented on wllvm/extraction.py:
Making readelf the default will break other platforms. You need to think about preserving behavior not just changing behavior. Again I am not going to merge in changes that are hacky or break support of other platforms.

I'm not sure that the  EXCLUDE_PATHS hack is needed. Why not just extract the bitcode out as a module and then do the linking yourself by hand?

ianamason commented:
I added another comment about the file type, but it seems to have disappeared, so here it is again.

I'm not going to merge in hard coded paths. When we use a tool (via Popen) we often provide the ability to include its location via a environment variable. This approach would work for 'file' but it would need to be documented. Something like WLLVM_FILE_TOOL_PATH
